verse.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const Chord = require('./chord');
+const voicePattern = /^([a-zA-Z-_]+)([0-9]*):\s(.*)/;
+
+function tokenize(instrument, voice, data) {
+  const re = /[^\s]+/g;
+  const events = [];
+
+  let match;
+  while (match = re.exec(data)) {
+    events.push({
+      index: match.index,
+      voice: voice,
+      content: instrument == 'c' ? Chord.parse(match[0]) : match[0]
+    });
+  }
+
+  return events;
+}
+
+function parseVoice(voice) {
+  const match = voice.match(voicePattern);
+  if (!match) {
```

Comment:
Since we're relying on `match` to have four elements, can we check that `match.length !== 4` here as well?

Reply:
There doesn't seem to be any way to produce a `match.length` other than 4. If the pattern doesn't match, then `match` is always `null`. If the pattern matches, then it is always of length 4.